src/services/wido.ts

```diff
@@ -0,0 +1,233 @@
+import { getAddress } from "@ethersproject/address";
+import { TransactionRequest } from "@ethersproject/providers";
+import { approveForZap, getBalances, getSupportedTokens, getTokenAllowance, getWidoContractAddress, quote } from "wido";
+
+import { Chains } from "../chain";
+import { Service } from "../common";
+import { EthAddress, SUPPORTED_ZAP_OUT_TOKEN_SYMBOLS, usdc, ZeroAddress } from "../helpers";
+import { Address, Balance, Integer, Token, TokenAllowance } from "../types";
+
+export class WidoService extends Service {
+  async supportedTokens(): Promise<Token[]> {
+    // unsupported networks
+    if (this.chainId === 1337) {
+      throw new Error("Unsupported");
+    }
+    const network = Chains[this.chainId];
+    const tokenList = await getSupportedTokens(this.chainId);
+
+    return tokenList
+      .filter((token) => !!token.symbol) // TODO(wido)
+      .map((token) => {
+        const address = token.address === ZeroAddress ? EthAddress : getAddress(token.address);
+
+        return {
+          address,
+          decimals: String(token.decimals),
+          icon: `https://assets.yearn.network/tokens/${network}/${token.address.toLowerCase()}.png`,
+          name: token.symbol,
+          priceUsdc: usdc("0"), // TODO(wido)
+          dataSource: "wido",
+          supported: {
+            widoZapIn: true,
+            widoZapOut: SUPPORTED_ZAP_OUT_TOKEN_SYMBOLS.includes(token.symbol.toUpperCase()), // TODO: add native token for multichain zaps
```

Comment:
Initially we are not supporting multichain, would we still need a TODO?
Can't we just say `true` here?

Reply:
> Initially we are not supporting multichain, would we still need a TODO?

This comment was from Gambitox in portals.ts, I just copied it, I'm not sure what it means. I removed it.
> Can't we just say true here?

I think yearn wants support only for a predefined list of tokens: stables, native token and btc.